### DISCUSSION TITLE
How to annotate schema with directives that are understood by graphql-java

### DISCUSSION THREAD

**Discussion by fediazgon** [graphql-java/graphql-java#3380] [Q&A]
`graphql-java` added support for the `@oneOf` directive [in the 21.2 release](https://github.com/graphql-java/graphql-java/releases/tag/v21.2). I understand that `graphql-kotlin` is still using the `21.1` release, but I was wondering if I will be able to use this directive out-of-the-box.

In the docs I can see that other directives like `@skip` are supported:

However, how do I annotate my schema and use `graphql-kotlin-schema-generator` to generate this annotations for me? What I want is to write Kotlin code like this:

```kotlin
@GraphqlOneOf
data class Some(
 val a: String?
 val b: Int?
)
```

and `graphql-kotlin-schema-generator` will generate code like:

```graphql
type Some @oneOf { ... }
```

Do I have to use custom directives for this?

**Comment by dondonz** [comment]
Hi @fediazgon I'm not very familiar with the graphql-kotlin library and I also don't know about this schema generator tool. Could you please ask in those repos for their advice?

**Comment by fediazgon** [comment]
Omg, I thought I was writing there. Wrong tab I guess.
